src/isotools/_transcriptome_io.py

```diff
@@ -411,41 +429,60 @@ def add_sample_from_bam(self, fn, sample_name=None, barcode_file=None, fuzzy_jun
                     #        if qual:
                     #            tr['mutations'][sample_name][pos][alt][1].append(qual) #assuming the quality already accounts for cov>1
 
+                    # checks if there is any "clipping" in the alignment, which occurs when part of the read doesn't align to the reference genome
+                    # CIGAR string in a SAM/BAM file represents the operations performed on each aligned segment of the read:
+                    #   '4' corresponds to "soft clipping," which means that the aligned segment is present in the SEQ but does not align to the reference
                     if 4 in read.cigartuples:  # clipping
                         s_name = sample_name if not barcodes else barcodes[tags['XC']]
                         clip = get_clipping(read.cigartuples, read.reference_start)
                         tr.setdefault('clipping', {}).setdefault(s_name, {}).setdefault(clip, 0)
                         tr['clipping'][s_name][clip] += cov
+                
                 for tr_interval in transcripts:
                     tr = tr_interval.data
                     tr_ranges = tr.pop('range')
                     # tr_ranges=tr['range']
+
+                    # get start and end positions of the transcript, and update the starts and ends dictionaries with coverage information
                     starts, ends = {}, {}
                     for r, cov in tr_ranges.items():
                         starts[r[0]] = starts.get(r[0], 0) + cov
                         ends[r[1]] = ends.get(r[1], 0) + cov
+                    
                     # get the median TSS/PAS
                     tr['exons'][0][0] = get_quantiles(starts.items(), [0.5])[0]
                     tr['exons'][-1][1] = get_quantiles(ends.items(), [0.5])[0]
+                    
+                    # calculate total coverage
                     cov = sum(tr_ranges.values())
```

Comment:
This comment isn't really needed, no? It's a sum, what else than `total` should it do?